platforms/core-runtime/base-annotations/src/main/java/org/gradle/internal/service/scopes/ServiceScope.java

```diff
@@ -36,6 +40,10 @@
 @Inherited
 public @interface ServiceScope {
 
-    Class<? extends Scope> value();
+    /**
+     * One or more scopes in which the service is declared,
+     * from the longest lifecycle to the shortest.
```

Comment:
🤔 What is happening if the condition `from the longest lifecycle to the shortest` is not satisfied? Is it a mandatory thing?
I'm agree, that it's definitely worth to declare scopes in a such ordering, but it seems like validation of that is missing. Do we want to have it?

Reply:
I would keep it as a recommendation for now, rather than a requirement. That's why I didn't add a validation. 

I think we need to first figure out what do we do with all service registries that still miss a scope (the right scope does not yet exist in `Scopes`). If we decide to introduce more scopes, then the ordering rules may get more complicated.

I'd rather see us working in the direction of declaring a service in exactly one scope.